### PR TITLE
Dummy note generation

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

File: src/note.rs [hunk 1]
```diff
@@ -53,6 +60,25 @@ pub struct Note {
...
+    /// Defined in [Zcash Protocol Spec § 4.8.3: Dummy Notes (Orchard)][orcharddummynotes].
+    ///
+    /// [orcharddummynotes]: https://zips.z.cash/protocol/nu5.pdf#orcharddummynotes
+    pub(crate) fn dummy(rng: &mut impl RngCore, rho: Option<Nullifier>) -> (FullViewingKey, Self) {
+        let fvk: FullViewingKey = (&SpendingKey::random(rng)).into();
+        let recipient = fvk.default_address();
+
+        let note = Note {
+            recipient,
+            value: NoteValue::default(),
```
Comment (nuttycom) [inline src/note.rs:74]: I really dislike the use of `default` here; can this be `zero()`?

**Review by nuttycom — COMMENTED** [review]
Minor complaint about the use of `default()`

**Comment by str4d** [comment]
I do not understand GitHub's PR-closing logic for chained PRs *sigh*

File: src/note.rs (hunk 1 above)
Comment (str4d) [reply src/note.rs:74]: Sure, I can add an alias.

**Review by nuttycom — APPROVED** [review]

**Comment by daira** [comment]
> str4d deleted the branch main yesterday
(It's not deleted, I checked. lol)

> I do not understand GitHub's PR-closing logic for chained PRs _sigh_
Yeah, it's kind of a head scratcher.

File: src/note/nullifier.rs [hunk 2]
```diff
@@ -12,6 +14,22 @@ use crate::{
...
+    /// the accompanying spent note within the action, which is constrained by consensus
+    /// to be unique. In the case of dummy spent notes, we get this restriction by
+    /// following the chain backwards: the nullifier of the dummy spent note will be
+    /// constrained by consensus to be unique, and the nullifier's uniqueness is derived
+    /// from the uniqueness of $\rho$.
+    ///
+    /// Instead of explicitly sampling for a unique nullifier, we rely here on the size of
+    /// the base field to make the chance of sapling a colliding nullifier negligible.
+    pub(crate) fn dummy(rng: &mut impl RngCore) -> Self {
+        Nullifier(pallas::Base::random(rng))
```
Comment (daira) [inline src/note/nullifier.rs:30]:
```suggestion
        Nullifier(extract_p(&pallas::Point::random(rng)))
```
(needs an import of `group::Group`).

**Comment by str4d** [comment]
Rebased on main at @daira's request.

File: src/note/nullifier.rs [hunk 3]
```diff
@@ -1,5 +1,7 @@
+use ff::Field;
```
Comment (daira) [inline src/note/nullifier.rs:1]:
```suggestion
use group::Group;
```
Part of the suggested change below.

**Review by daira — APPROVED** [review]
utACK modulo comment.

**Review by daira — APPROVED** [review]
utACK 4db3b54c8b5bd9fca6d9b655c47374898a6a72f9

**Pull request merged**